### DISCUSSION TITLE
Scroll restoration bug?

### DISCUSSION THREAD

**Discussion by l-you** [vercel/next.js#54268] [Help] [answered]
### Summary

Please look at the following IOS bug on screen recording.
Swiping up and navigating using "back" button breaks scroll position of previous page.
Its reproducible on any website including https://nextjs.org.

Both browser 'auto' scroll restoration and `scrollRestorration` config option does not help.
It's really annoying on e-commerce websites where there can be 80 product items. 

### Example

https://github.com/vercel/next.js/assets/32598874/51d9839f-f129-4a6d-9a7a-f2013c2de45d

**Comment by shunshimono** [comment]
同じ問題を抱えています。

**Reply by l-you** [reply, marked as answer]
You can try https://github.com/RevoTale/next-scroll-restorer to fix your issue
